### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by bobb-Rob**
### I Implemented the following: 

- Create a app.rb file that will serve as your console app entry-point. It should have methods that do the following:

     - List all books.
     - List all people.
     - Create a person (teacher or student, not a plain Person).
     - Create a book.
     - Create a rental.
     - List all rentals for a given person id.

- In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:

    - Present the user with a list of options to perform.
    - Lets users choose an option.
    - If needed, ask for parameters for the option.
    - Have a way to quit the app.

### CODE REVIEW

File: classes/app.rb [hunk 1]
```diff
@@ -0,0 +1,159 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+
+class App
```
Comment (rustam-tolipov) [inline classes/app.rb:9]: You made a great effort to make this project! However, some errors need to be filled in!

File: classes/app.rb [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  end
+
+  private
+
+  # create new student method
+  def create_student
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp
```
Comment (rustam-tolipov) [inline classes/app.rb]: If I give to student's age a wrong argument, it is not rising an error. 
You can see the image here:

Screen Shot 2022-08-26 at 21 39 06

- [ ] Kindly fix this issue. When you give a wrong argument, make sure it shows an error!

File: classes/app.rb [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    puts "\n"
+    puts 'New student created successfully'
+  end
+
+  # create new teacher method
+  def create_teacher
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp
```
Comment (rustam-tolipov) [inline classes/app.rb]: [OPTIONAL] The same error as I mentioned above!

File: classes/app.rb [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    marked_book = gets.chomp.to_i
+    puts "\n"
+    puts 'Select a person from the following list by number (not id)'
+    puts "\n"
+    @people.each_with_index do |person, index|
+      puts "No. #{index + 1}) - Name : #{person.name} |ID : #{person.id} | Age : #{person.age}"
+    end
+    puts "\n"
+    rental_person = gets.chomp.to_i
+    print 'Date (YYYY/MM/DD) : '
```
Comment (rustam-tolipov) [inline classes/app.rb]: - [ OPTIONAL] The same error is present here:


Screen Shot 2022-08-26 at 21 38 55

File: classes/main.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
+require_relative 'app'
+
+class TempMain
+  def self.display_options
+    puts 'Please choose an option by entering a number:
+    1 - List all books.
+    2 - List all people.
+    3 - Create a person (teacher or student, not a plain Person).
+    4 - Create a book.
+    5 - Create a rental.
```
Comment (rustam-tolipov) [inline classes/main.rb:10]: When I tried to create a rental, it broke the app:

Screen Shot 2022-08-26 at 21 37 17

- [ ] Make sure all errors are handled!

File: classes/main.rb [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      puts 'Enter a number between 1 and 7'
+    end
+  end
+end
+
+def main
+  TempMain.new
+end
+
+main
```
Comment (rustam-tolipov) [inline classes/main.rb:49]: [OPTIONAL] Kindly make sure you added MIT license to your project.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi @bobb-Rob, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @bobb-Rob , 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well implemented suggested changes.

### Optional suggestions :arrow_down_small: 

- [ ] The commit message should be written in the imperative tense. `Ex: "Fix bug" instead of "Fixed bug".` [For more details.](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**